flatland/callbacks/callbacks.py

```diff
@@ -1,10 +1,12 @@
 from pathlib import Path
-from typing import Optional, List
+from typing import Optional, TypeVar, Generic
 
 from flatland.core.env import Environment
 
+EnvironmentType = TypeVar('EnvironmentType', bound=Environment, covariant=True)
 
-class FlatlandCallbacks:
+
+class FlatlandCallbacks(Generic[EnvironmentType]):
```

Comment:
Future refactoring: should/can callbacks and trajectories go in same package and be implemented in the core?

Reply:
https://github.com/flatland-association/flatland-rl/issues/277